### ISSUE TITLE
catalog-info.yaml file still exists error file - Register an existing component

### ISSUE THREAD

**Issue by Yashit-G** [backstage/backstage#22669]
Hi Everyone I was trying to register an existing component in backstage application.
The error that I had on the second step of creating the catalog file was resolved after reading this issue: https://github.com/backstage/backstage/issues/3811

So currently my second step gets completed smoothly and I am on the 3rd step of review

But now on the  REVIEW step it gives me this error which I don't understand why

{"error":{"name":"ConflictError","message":"Location url:https://github.com/Navdeep-12/Backstage/blob/main/catalog-info.yaml already exists","stack":"/Users/navdeep/Documents/backstage-master/plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts:49\n throw new ConflictError(\n ^\n\nConflictError: Location url:https://github.com/Navdeep-12/Backstage/blob/main/catalog-info.yaml already exists\n at <anonymous> (/Users/navdeep/Documents/backstage-master/plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts:49:15)\n at runMicrotasks (<anonymous>)\n at processTicksAndRejections (node:internal/process/task_queues:96:5)"},"request":{"method":"POST","url":"/locations"},"response":{"statusCode":409}}

It says my catalog-info.yaml file exists in the above path, I have checked multiple times that there is no such file in the given path, 

Moreover In the FIRST STEP OF URL I provide this link : -
https://github.com/Navdeep-12/Backstage/tree/main/packages/catalog-model/examples/components
It should be creating a yaml file in there why does it goes and find the above path for which it is throwing error.
![Screenshot 2024-02-03 055044](https://user-images.githubusercontent.com/116294051/302000503-7d29343a-b04d-47f7-88f4-9df19c4ab01d.png)
So I don't understand why this error comes 
            
```[tasklist]
### Tasks
```

**Comment by mohamadalsalty** [comment]
Hello,

I encountered the same issue.

The problem was resolved by importing the component using the full URL of catalog-info.yaml.

Example: https://github.com/XXX/XXX/blob/master/catalog-info.yaml.

**Comment by Yashit-G** [comment]
I tried giving this but now the I am again at the same bug
I have created a config-info.yaml file and that is empty because I want to add the metadata into the file in second step only

Now when I provide the URL like this  : https://github.com/XXX/XXX/blob/master/catalog-info.yaml.
While creating PR  it gives me errors like this: 
Couldn't create a commit with catalog-info.yaml file added, received http response status code 422: Invalid request. "sha" wasn't supplied.

Which takes me back to the same story of this : - https://github.com/backstage/backstage/issues/3811

I dont understand what to do now

**Comment by Jokas377** [comment]
I am having these exact same issues and have been searching for a solution. I had a repo with a catalog-info.yaml file in it, but when I tried to link the repo, it would try to add another catalog-info.yaml file as if it could not find the pre-existing one. I then got the 422 error you received when trying to send the pull request.

When putting the full url for the catalog-info.yaml file, I got the same long error you received. I tried deleting my catalog-info.yaml file and adding a new one through the "register existing component", but I would still receive the 422 error as if the file was still there.
